### ISSUE TITLE
Thinking about the client hardware

### ISSUE THREAD

**Issue by gene96817** [ann0see/rasjam#3]
In exploring this repository, it seems you have decided that all musicians/clients will use RPi. 

What led to the conclusion not to use any Mac or Windows machines that the musician already owns?
I do agree that Window's users are very difficult to onboard to Jamulus. The audio interface issue stumps most Windows users.

What about musicians that have high-quality mics and audio-interfaces? 
Are you expecting the musicians to only use USB mics with the RPi?

I am looking for people that have experience running Jamulus on Chromebooks. A lot of public schools around here have given students Chromebooks. I don't have any experience (yet) on this. I need a lot of help on this.

**Comment by ann0see** [comment]
Hi! I really appreciate your answer. I'm sorry for the messy repo. 

We were about 3 people at the beginning but since in my country the musicians are allowed to do live rehearsals in small groups they no longer actively participate. 

I'll try to comment on the problems/ideas you are raising (but I will probably not be able to do everything now). 

> What led to the conclusion not to use any Mac or Windows machines that the musician already owns?

Since
* I can't assume that everybody knows how to install a program
* Not every student has access to PC yet (although they mostly have Smartphones)
* Setup is much easier for the student and they have a Plug and Play experience (since everything was setup for him)
* if somebody can setup jamulus on their own pc of course, he's free to do so

I must assume that the student has almost nothing at home and his parents don't know how to use their pc for everything other than surfing the Internet/checking E-Mails. 
I know that's a quite complex problem but unfortunately reality. Some even don't have DSL; only - if we're lucky - a smartphone.

**Comment by gene96817** [comment]
Hi… Where are you located? What country are you?
What kind of PCs do you think are common in your area? The CPU power matters.
If you assume your musicians have PCs then booting from a USB stick might be a good solution.

I think a Chromebook might be about the same cost as a RPi4. I don’t know if an RPi2 is sufficient as a client. I will be working on understanding how to use a Chromebook.

GEne
--------------------------
Eugene Chang
gene96817@gmail.com
eugene.chang@alum.mit.edu
781-799-0233



************
What is the attached file signature.asc?
If you use PGP compatible mail software, this file holds a cryptographic digital signature to prove the email was really sent by me. As a bonus, if you are using PGP, our email could be cryptographically secure and private from anyone intercepting the email.

**Comment by ann0see** [comment]
I don't think the cpu power on the client matters (jamulus runs well on my Raspberry Pi 3B).

I can send you an e-mail but would prefer to handle most on GitHub. 

In my country chromebooks are not common. Schools often use iPads or Windows laptops.

I live in Europe.

**Comment by gene96817** [comment]
Have you watched any of the Jamulus World Jam on YouTube? I want to deliver a similar experience for music classes with 15-20 students. The schools or the local community drives the choice of devices for the students. Cost matters. I think it is hard to ask some schools and students to buy a RPi for each student. A RPi3B doesn’t cost much but it will need a special decision to budget the money for music students. Here in the US, music education is not well funded at many schools.

I would be interested in a group woking on delivering easy to use Jamulus on the devices already selected by the schools. The World Jam experience requires running Jamulus and Zoom. I am testing an open source videoconferencing (instead of Zoom) so that it does not add more cost to the schools. This does mean a headless RPi is not adequate.

Why video? Musicians watch at each other or watch the director for timing. For education, the music teacher will want to see the student/musicians. It is also important because few music styles keep strict time. Only expert musicians can keep a small ensemble together purely by ear and the presence of delay magnifies the challenges of performing in sync.

Presently I am working on an easy to install server for the music teacher. I don’t know anyone who has delivered (or is working on) Jamulus for Chromebooks or iPads. Here in Hawaii, Chromebooks are the standard at the public schools.

Gene
--------------------------
Eugene Chang
gene96817@gmail.com
eugene.chang@alum.mit.edu
781-799-0233



************
What is the attached file signature.asc?
If you use PGP compatible mail software, this file holds a cryptographic digital signature to prove the email was really sent by me. As a bonus, if you are using PGP, our email could be cryptographically secure and private from anyone intercepting the email.

**Comment by ann0see** [comment]
> Have you watched any of the Jamulus World Jam on YouTube?

Yes I have. And it's stunning that something like this is possible ;-).

> Cost matters

Yeah. But probably less for my than for your use case. In my case the school has USB-Audio interfaces, microphones and probably the budget to buy Raspberry Pis.

I'll probably also ditch the headless idea since it adds a lot of other problems (level balancing etc.).
Jamulus lacks moderating tools like general muting, kicking,... but I've already coded something on my jamulus fork (branch eduTools). I don't know if it's needed.

Considering your server idea: Why don't you setup an own central server and register servers on it like world jam?
> Jamulus for Chromebooks or iPads

Chromebooks might be possible @gilgongo runs on a chromebook (as far as I know) but I'm not sure if he runs jamulus on it. iPads probably won't work.

**Comment by gene96817** [comment]
Comments inline below.

Gene
--------------------------
Eugene Chang
gene96817@gmail.com
eugene.chang@alum.mit.edu
781-799-0233



************
What is the attached file signature.asc?
If you use PGP compatible mail software, this file holds a cryptographic digital signature to prove the email was really sent by me. As a bonus, if you are using PGP, our email could be cryptographically secure and private from anyone intercepting the email.


> On Oct 12, 2020, at 9:13 AM, ann0see <notifications@github.com> wrote:
> 
> 
> Have you watched any of the Jamulus World Jam on YouTube?
> Yes I have. And it's stunning that something like this is possible ;-).
> 
> 
If my demo works, the answer is yes.
My first goal is to support music education, the Virtual Rehearsal Room.
I can dream up a slightly different use case where the VRR is a virtual coffee house with musicians interacting with the audience. Ha, imagine this also becoming a comedy club.

> Cost matters
> Yeah. But probably less than for your use case. In my case the school has USB-Audio interfaces, microphones and probably the budget to buy Raspberry Pis.
> 
I’ll have to use RPi if Chromebooks are impossible.

> I'll probably also ditch the headless idea since it adds a lot of other problems (level balancing etc.).
> Jamulus lacks moderating tools like general muting, kicking,... but I've already coded something on my jamulus fork (branch eduTools). I don't know if it's needed.
> 
I don’t want to spend a lot of time on these details until the demo is running.
However, I have been thinking about the UI a director needs for a group of 10-20 musicians. The current UI will need improvements for a director.


> Considering your server idea: Why don't you setup an own central server and register servers on it like world jam?
> 
I am concerned that the Jamulus community in dropping in on each other to jam as the JamKazam community. A private server  central server is trivial but it will not build a community that shares music.

> Jamulus for Chromebooks or iPads
> 
> Chromebooks might be possible @gilgongo <https://github.com/gilgongo> runs on a chromebook (as far as I know) but I'm not sure if he runs jamulus on it. iPads probably won't work.
> 
> 

I won’t put any effort on iPads for at least a year. Apple has a plan to unify MacOS and IOS. This means MacOS applications will run on IOS and vise versa. The devil is in the details. Any work now might be wasted.
